### PR TITLE
WIP: Fix IRGen for associated type witness accessor functions

### PR DESCRIPTION
**Pull request by aschwaighofer**
in generic contexts with same type requirements.

rdar://27718494

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test os x platform

File: lib/IRGen/GenProto.cpp [hunk 1]
```diff
@@ -1276,7 +1280,18 @@ class AccessorConformanceInfo : public ConformanceInfo {
       }
       return *Fulfillments;
     }
-  };
+
+    void bindArchetypeAccessPathsInConformance(IRGenFunction &IGF) {
+      auto TheDecl = Conformance.getDeclContext();
```
Comment (rjmccall) [inline lib/IRGen/GenProto.cpp]: The standard abbreviation for a DeclContext is DC.  Please don't call it a Decl; that's actively somewhat misleading.

Comment (rjmccall) [inline lib/IRGen/GenProto.cpp]: Also, I know this is a third rail that nobody wants to touch, but IRGen is pretty consistent about using lowercase identifiers for local variables.

**Comment by rjmccall** [comment]
A couple of minor style comments, but otherwise LGTM.

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test os x platform

**Pull request merged**